test/pylib/scylla_cluster.py

```diff
@@ -0,0 +1,792 @@
+        """Start a stopped server"""
+        logging.info("Cluster %s starting server", self)
+        if server_id in self.running:
+            return True, f"Server {server_id} already started"
+        if server_id in self.removed:
+            return False, f"Server {server_id} removed"
+        if server_id not in self.stopped:
+            return False, f"Server {server_id} unknown"
+        self.is_dirty = True
+        server = self.stopped.pop(server_id)
+        if self.last_seed is None:
+            self.last_seed = server.host
+        server.seed = self.last_seed
+        await server.start()
+        self.running[server_id] = server
+        return True, f"Server {server_id} started"
+
+    async def server_restart(self, server_id: str) -> Tuple[bool, str]:
+        """Restart a running server"""
+        logging.info("Cluster %s restarting server %s", self, server_id)
+        if server_id in self.running:
+            self.is_dirty = True
+            server = self.running.pop(server_id)
+            await server.stop_gracefully()
+            await server.start()
+            self.running[server_id] = server
+            return True, f"Server {server_id} restarted"
+        if server_id in self.stopped:
+            self.is_dirty = True
+            server = self.stopped.pop(server_id)
+            await server.start()
+            self.running[server_id] = server
+            return True, f"Server {server_id} started"
+        if server_id in self.removed:
+            return False, f"Server {server_id} removed"
+        return False, f"Server {server_id} unknown"
+
+    async def server_remove(self, server_id: str) -> Tuple[bool, str]:
+        """Remove a specified server"""
+        self.is_dirty = True
+        logging.info("Cluster %s removing server %s", self, server_id)
+        if server_id in self.running:
+            server = self.running.pop(server_id)
+            await server.stop_gracefully()
+            self.update_last_seed(server.host)
+        elif server_id in self.stopped:
+            server = self.stopped.pop(server_id)
+        else:
+            return False, f"Server {server_id} unknown"
+        await server.uninstall()
+        self.removed.add(server_id)
+        return True, f"Server {server_id} removed"
+
+
+class Harness:
+    """Manages a pool of Scylla clusters for running test cases"""
+    def __init__(self, test_name: str, base_dir: str, clusters: Pool[ScyllaCluster]) -> None:
+        self.test_name: str = test_name
+        self.clusters: Pool[ScyllaCluster] = clusters
+        self.cluster: Optional[ScyllaCluster] = None  # Current cluster
+        self.is_before_test_ok: bool = False
+        self.is_after_test_ok: bool = False
+        self.is_running: bool = False
+        # API
+        # NOTE: need to make a safe temp dir as tempfile can't make a safe temp sock name
+        self.harness_dir: str = tempfile.mkdtemp(prefix="harness-", dir=base_dir)
```

Comment:
Now that we create an instance of harness for each test, to simplify debugging it would be best in my view to derive the name of the socket from the test name. something like test-name.test-id.sock.
Please note that this file needs to be removed in the end, so please create an artifact for it.

Reply:
As topology tests require Scylla and Raft, I removed the `this_dc` hack from topology and use normal replication for keyspaces there. Other test suites are unchanged.